backend/layers/persistence/persistence.py

```diff
@@ -17,80 +21,197 @@
     DatasetVersion,
     DatasetVersionId,
 )
+from backend.layers.persistence.orm import (
+    Collection as CollectionRow,
+    CollectionVersion as CollectionVersionRow,
+    Dataset as DatasetRow,
+    DatasetVersion as DatasetVersionRow,
+    DatasetArtifact as DatasetArtifactRow
+)
+from backend.layers.persistence.db_session import db_session_manager
+from sqlalchemy import select
+
+from backend.layers.persistence.persistence_interface import DatabaseProviderInterface
+
+
+class DatabaseProvider(DatabaseProviderInterface):
 
+    def __init__(self) -> None:
+        self.db_session_manager = db_session_manager
+
+    @staticmethod
+    def _generate_id():
+        return str(uuid.uuid4())
+
+    def _hydrate_dataset_version(self, dataset_version: DatasetVersionRow) -> DatasetVersion:
+        """
+        Populates canonical_dataset, artifacts, and status for DatasetVersionRow
+        """
+        if not dataset_version.canonical_dataset:
+            dataset_version.canonical_dataset = self.get_canonical_dataset(dataset_version.dataset_id)
```

Comment:
my understanding is that a relationship in ORM is basically an auto-join run on every query to a table, and we're trying to avoid JOINs for reasons outlined in the Portal Data Management Redesign Tech Spec. We're purposely moving towards lookup table model